src/GovUk.Education.ExploreEducationStatistics.Content.Services/Cache/PublicationCacheService.cs

```diff
@@ -24,9 +32,94 @@ public PublicationCacheService(IPublicationService publicationService)
         return _publicationService.Get(publicationSlug);
     }
 
+    public async Task<Either<ActionResult, IList<ThemeTree>>> GetPublicationTree(
+        PublicationTreeFilter filter)
+    {
+        var fullPublicationTree = await GetFullPublicationTree();
+
+        return await fullPublicationTree
+            .ToAsyncEnumerable()
+            .SelectAwait(async theme => await FilterThemeTree(theme, filter))
+            .Where(theme => theme.Topics.Any())
+            .OrderBy(theme => theme.Title)
+            .ToListAsync();
+    }
+
     [BlobCache(typeof(PublicationCacheKey), forceUpdate: true, ServiceName = "public")]
     public Task<Either<ActionResult, PublicationCacheViewModel>> UpdatePublication(string publicationSlug)
     {
         return _publicationService.Get(publicationSlug);
     }
+
+    [BlobCache(typeof(PublicationTreeCacheKey), forceUpdate: true, ServiceName = "public")]
+    public Task<IList<ThemeTree>> UpdatePublicationTree()
+    {
+        _logger.LogInformation("Updating cached Publication Tree");
+        return _publicationService.GetPublicationTree();
+    }
+
+    [BlobCache(typeof(PublicationTreeCacheKey), ServiceName = "public")]
+    private Task<IList<ThemeTree>> GetFullPublicationTree()
```

Comment:
I was expecting to see the class "ThemeTree" renamed to "PublicationTree"

Reply:
Thanks, I've renamed things now to reduce the confusion.

`ThemeTree` is now `PublicationTreeThemeViewModel`
`TopicTree` is now `PublicationTreeTopicViewModel`
`PublicationTreeNode` is now `PublicationTreePublicationViewModel`

Associated method names and variable names have been updated on the same commit, thanks. 🙂